### PR TITLE
Add tx_search endpoint for RPC client

### LINKED ISSUE #698: tx_search RPC
IBC requires the `tx_search()` endpoint in order to retrieve pending IBC packets. I was not able to find one.
The Go implementation is here:
https://github.com/tendermint/tendermint/blob/095e9cd7ef0e59c8b716a3bde32ca5b7a00f8e30/rpc/client/http/http.go#L465-L491

### PR DESCRIPTION
**Pull request by thanethomson**
Closes #698 

* [x] Referenced an issue explaining the need for the change
* [x] Updated all relevant documentation in docs
* [x] Updated all code comments where relevant
* [x] Wrote tests
* [x] Updated CHANGELOG.md

### CODE REVIEW

**Review by greg-szabo — COMMENTED** [review]
Generally looks good.
I'm happy to see the correct use of our slightly messy serialization solution :) and I'm even happier to see rpc-probe in action.

Tokio might need some more cleanup or fixes according to the CI. I'll run through it again once the CI issues are fixed.

File: tendermint/tests/integration.rs [hunk 1]
```diff
@@ -351,4 +349,75 @@ mod rpc {
...
+        assert_eq!(tx_info.tx, txs[0].tx.as_bytes());
+
+        subs_client.close().await.unwrap();
+        driver_handle.await.unwrap().unwrap();
+    }
+
+    async fn broadcast_tx(
+        http_client: &HttpClient,
+        websocket_client: &mut WebSocketClient,
+        key: &str,
```
Comment (melekes) [inline tendermint/tests/integration.rs:399]: why not pass tx as a whole `broadcast_tx(http_client, websocket_client, tx)`?

**Review by melekes — APPROVED** [review]
👍

File: tendermint/tests/integration.rs (hunk 1 above)
Comment (thanethomson) [reply tendermint/tests/integration.rs:399]: Ah, this is a decision from a previous iteration. I'll change this 👍

**Comment by ancazamfir** [comment]
Looks good, I was able to get the tx info for IBC packets! Some small issue with the event key/ value encoding but orthogonal.
Thanks Thane!!

**Review by ancazamfir — APPROVED** [review]
Haven't looked at the code in a lot of detail but I was able to integrate with this in IBC so for me it's good to go 👍  Thanks Thane!!

**Review by ancazamfir — APPROVED** [review]

**Pull request merged**